### PR TITLE
feat: product warranties

### LINKED ISSUE #1062: Product Warranties
Implement Product Warranties in the PWA similar to the Responsive Starter Store (see http://pwa-public-demo.test.intershop.com/INTERSHOP/web/WFS/inSPIRED-inTRONICS-Site/en_US/-/EUR/ViewProduct-Start?SKU=5777062&CategoryName=1584&CatalogID=Home-Entertainment).

image

[AB#75155](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/75155)

### PR DESCRIPTION
**Pull request by shauke**
## PR Type

[x] Feature

## What Is the Current Behavior?

The PWA does not yet handle product warranties.

Issue Number: Closes  #1062

## What Is the New Behavior?

The user is able to buy a product together with a warranty. He/she can select a warranty on the product detail page and the cart page. The selected warranties are shown during the checkout process and on requisition and order detail pages.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

## Limitations

The PR requires an ICM 11.7 or higher.

With lower ICM versions the following issues are known:
If the user removes a warranty from a basket line item the basket calculation is not promptly triggered.
After submitting an order that contains a product with a warranty a general error page is shown. [AB#80443](https://dev.azure.com/intershop-com/Products/_workitems/edit/80443)
This causes the current e2e-test to fail.

Once the order process can be finished, additional e2e-tests for the receipt page and order-detail page have to be added.
These issues will be fixed in ICM 7.10.41.

If the user changes the shipping method the basket cannot be ordered any more due to limitations of the ICM REST interface.
To overcome this problem the ICM has to be configured to use [single shipment only](https://intranet.intershop.com/kb/index.php/Display/240R62#Cookbook-ShippingUseCases-Recipe:SetUpaSingleShipmentEnvironmentandItsLimitations).

## Other Information

IAD: [https://www.figma.com/file/4ApE1rE1fsjaKyrUtd90Rd/Product-Warranties?node-id=0%3A1](https://www.figma.com/file/4ApE1rE1fsjaKyrUtd90Rd/Product-Warranties?node-id=0:1)

Pending UI Issues:
- [ ] display warranty name (isntead of id) in the basket items overview of the checkout side bar
- [ ] display single warranty price in the line items list

[AB#83337](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/83337)

### CODE REVIEW

**Comment by iwiederhold** [comment]
alignement_warranties

**Comment by iwiederhold** [comment]
remove_border

**Review by iwiederhold — CHANGES_REQUESTED** [review]
See comments.

**Comment by schadii** [comment]
In the detail modal, the phrase “Insurance against breakdown is repeated. Warranty period: 1 year.” twice. It may be clearer to mention it only once and then list the specific details under it. 

Please consider using bullet points or a numbered list for better readability, especially since you are detailing specific information. This can make it easier for the user to absorb the information.

The term "Expires: Service Plan" might be a bit confusing. Could not consider specifying the exact expiration date of the warranty or provide details on how the user can find this information?
Detail modal

**Review by schadii — APPROVED** [review]

**Review by iwiederhold — APPROVED** [review]

File: src/assets/i18n/en_US.json [hunk 1]
```diff
@@ -1038,6 +1038,13 @@
   "product.short_description.label": "Short Description",
   "product.specific_attributes.label": "Specific Attributes",
   "product.variations.text": "Variations",
+  "product.warranty.code.text": "Warranty Code:",
+  "product.warranty.detail.text": "Details",
+  "product.warranty.expire_date.text": "Expires:",
+  "product.warranty.heading.text": "Protect your purchase:",
```
Comment (mglatter) [inline src/assets/i18n/en_US.json:1044]: Sounds like it was insecure to buy anything in our online shop. Would suggest rephrasing it to "Protect your purchased product" or some other headline that makes clear that this is about a warranty for products.

File: src/assets/i18n/en_US.json [hunk 2]
```diff
@@ -1038,6 +1038,13 @@
...
+  "product.warranty.no_protection_plan": "No protection plan",
```
Comment (mglatter) [inline src/assets/i18n/en_US.json:1045]: What is meant by protecttion plan? Is it the same as the service plan mentioned in the modal Shadi commented on?

File: src/assets/i18n/de_DE.json [hunk 3]
```diff
@@ -1038,6 +1038,13 @@
   "product.short_description.label": "Kurzbeschreibung",
   "product.specific_attributes.label": "Spezifische Attribute",
   "product.variations.text": "Variationen",
+  "product.warranty.code.text": "Garantie-Code:",
+  "product.warranty.detail.text": "Details",
+  "product.warranty.expire_date.text": "Läuft aus am:",
+  "product.warranty.heading.text": "Schützen Sie Ihren Einkauf:",
+  "product.warranty.no_protection_plan": "Kein Schutzplan",
```
Comment (mglatter) [inline src/assets/i18n/de_DE.json:1045]: "Schutzplan" sounds quite dramatic. Is this the correct term to use in context of product warranties? Browsing the net, I'm not so sure about it.

File: src/assets/i18n/en_US.json (hunk 1 above)
Comment (mglatter) [reply src/assets/i18n/en_US.json:1044]: Suggestion: "Product warranty options:",

File: src/assets/i18n/en_US.json (hunk 2 above)
Comment (mglatter) [reply src/assets/i18n/en_US.json:1045]: Suggestion: "No warranty",

**Comment by mglatter** [comment]
I will change the English wording to the suggestions agreed on with Shadi and will provide updated localization values for the other languages as well.

**Review by mglatter — APPROVED** [review]

**Pull request merged**